pysph/base/gpu_nnps_base.pyx

```diff
@@ -294,9 +294,9 @@ cdef class BruteForceNNPS(GPUNNPS):
 
     cdef void find_neighbor_lengths(self, nbr_lengths):
         arguments = \
-                """%(data_t)s* s_x, %(data_t)s* s_y, %(data_t)s* s_z,
-                %(data_t)s* s_h, %(data_t)s* d_x, %(data_t)s* d_y,
-                %(data_t)s* d_z, %(data_t)s* d_h, unsigned int num_particles,
+                """%(data_t)s* d_x, %(data_t)s* d_y, %(data_t)s* d_z,
```

Comment:
Good catch!  I think a small comment here and elsewhere is in order to make sure that someone reading the source remembers that the global work size is determined from the length of the first argument.